### ISSUE TITLE
replace setup.py with flit

### ISSUE THREAD

**Issue by Akuli** [Akuli/porcupine#35]
I like flit for putting stuff to pypi easily, but afaik flit didn't exist when I was getting started with porcupine.

**Comment by Akuli** [comment]
this issue is actually pretty dumb.

First of all, porcupine isn't on pypi, so only `pip install path/to/repo` and `pip install url-of-porcupine` actually matter. flit would work for that, although it's mostly for putting stuff to pypi.

flit or setup.py can't import porcupine for figuring out porcupine's version number because dependencies might not be installed yet. It seems like flit really wants to `import porcupine` though, so to use it, I would need to make `__init__.py` do less things. This would mean changing `from porcupine import get_main_window` to `from porcupine.something import get_main_window`. I don't like that.

Currently `setup.py` finds the version number (and some other things) from `__init__.py` with a regex. If that causes any problems in the future, `release.py` could be updated to instead keep the version number in two places, `porcupine/__init__.py` and `setup.py`.

`setup.py` checks that the Tcl/Tk version that tkinter is using is recent enough. If setup.py was replaced with something that doesn't support running arbitrary code, then that check would instead need to happen when porcupine starts. That's not too bad, but it might not be a great experience to install something only to find out that it doesn't work because something else is too old.

tl;dr: `setup.py` is fine for porcupine
